src/utils.js

```diff
@@ -283,11 +283,32 @@ function getModulesPlugins(options, loaderContext) {
       extractImports(),
       modulesScope({
         generateScopedName(exportName) {
-          return getLocalIdent(loaderContext, localIdentName, exportName, {
-            context: localIdentContext,
-            hashPrefix: localIdentHashPrefix,
-            regExp: localIdentRegExp,
-          });
+          let localIdent = getLocalIdent(
+            loaderContext,
+            localIdentName,
+            exportName,
+            {
+              context: localIdentContext,
+              hashPrefix: localIdentHashPrefix,
+              regExp: localIdentRegExp,
+            }
+          );
+
+          // A null/undefined value signals that we should invoke the default
+          // getLocalIdent method.
+          if (localIdent == null && getLocalIdent !== defaultGetLocalIdent) {
```

Comment:
Why we need `getLocalIdent !== defaultGetLocalIdent`?

Reply:
In the situation where there wasn't a custom `getLocalIdent` function defined and the default function returned `null` or `undefined` for some reason, I didn't want to run the default again. It wouldn't hurt anything to do so, but it also wouldn't help anything either! ;)

Reply:
I think it is bad logic, we don't need `getLocalIdent !== defaultGetLocalIdent`, you can't use `defaultGetLocalIdent` in own code

Reply:
Well, at [line 140](https://github.com/tvsbrent/css-loader/blob/96ba438dc19ab833966a0aad46fc00188e49e14d/src/utils.js#L140) in the `utils.js`, the `getLocalIdent` property defaults to `defaultGetLocalIdent` function. So if a user doesn't provide a function, they are implicitly using the default function. The intent for the `!== defaultGetLocalIdent` check is to handle the situation where the user didn't provide a `getLocalIdent` function and `defaultGetLocalIdent` returned a `null` or `undefined`.

That may never happen, as there may already be safeguards in place to keep that from happening - and even if it did run the same function twice, it probably wouldn't be a big deal - so that check can be removed.